lib/deps.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+function makeErrorModule(error) {
+  return new Proxy(
+    {},
+    {
+      get: () => {
+        throw error;
+      },
+      set: () => {
+        throw error;
+      }
+    }
+  );
+}
+
+function retrieveKerberos() {
```

Comment:
The only change I would consider in this file is memoizing each of the deps. `require` _should_ handle this for you, but it might be a safe bet to ensure it ourselves, and it saves a function call at least:

```
let Kerberos;
function retrieveKerberos() {
  if (Kerberos) {
    return Kerberos;
  }

  try {
    Kerberos = require('kerberos'); 
  } catch(err) {
    Kerberos = makeErrorModule(err);
  } 

  return Kerberos;
}
```

Another consideration, especially since you've made this Proxy, is that you can remove the need to dynamically load any of this. Pull the code out of the `retrieveXXX` methods into the global scope of the module, and export the values immediately at runtime.